src/components/sdk/PreviewVideo/index.tsx

```diff
@@ -21,39 +22,42 @@ const StyledPreview = styled(VideoTile)`
 
 export const PreviewVideo: React.FC<BaseSdkProps> = (props) => {
   const audioVideo = useAudioVideo();
+  const meetingManager = useMeetingManager();
   const videoEl = useRef<HTMLVideoElement>(null);
-  const { selectedDevice } = useVideoInputs();
-  const { isVideoEnabled, setIsVideoEnabled } = useLocalVideo();
 
+  const { setIsVideoEnabled } = useLocalVideo();
+  // TODO: Move this to the Video Input Provider and expose only one selected Video Input device state
+  const [device, setDevice] = useState<Device | VideoTransformDevice | null>(meetingManager.selectedVideoInputTransformDevice);
+
+  // TODO: Move this to the Video Input Provider and expose only one selected Video Input device state
   useEffect(() => {
-    if (!audioVideo || !selectedDevice || !videoEl.current) {
-      return;
-    }
-    let mounted = true;
     const videoElement = videoEl.current;
+    meetingManager.subscribeToSelectedVideoInputTransformDevice(setDevice);
+    return () => {
+      if (videoElement) {
+        audioVideo?.stopVideoPreviewForVideoInput(videoElement);
+      }
+      meetingManager.unsubscribeFromSelectedVideoInputTranformDevice(setDevice);
+      setIsVideoEnabled(false);
+    };
+  }, []);
 
+  useEffect(() => {
+    if (!audioVideo || !device || !videoEl.current) {
+      return;
+    }
     async function startPreview() {
       if (!audioVideo) {
         return;
       }
-
-      await audioVideo.chooseVideoInputDevice(selectedDevice);
-      if (videoEl.current && mounted) {
+      await meetingManager.selectVideoInputDevice(device);
+      if (videoEl.current) {
         audioVideo.startVideoPreviewForVideoInput(videoEl.current);
+        setIsVideoEnabled(true);
       }
     }
-
     startPreview();
-
-    return () => {
-      mounted = false;
-
-      if (videoElement) {
-        audioVideo.stopVideoPreviewForVideoInput(videoElement);
-        if (isVideoEnabled) setIsVideoEnabled(false);
-      }
-    };
-  }, [audioVideo, selectedDevice]);
```

Comment:
Should this still have `audioVideo` dependency?

Also, should we keep the start and the cleanup steps such as stopping preview for `PreviewVideo` component APIs in one hook and the transform device subscription in the first one?

Reply:
Yup, my bad. I can separate the subscription one, but the start and stop preview cant depend on the same useEffect dependencies. 

We should only call stopPreview when the audioVideo changes. The startPreview should be called when the device changes, but not when the audioVideo changes. I'll show you in the next revision.

Reply:
Yeah that is correct, but we would still need a valid `audioVideo` to call the `stopVideoPreviewForVideoInput` API hence thought so, to avoid calling startPreview on audioVideo changes we can check against isVideoEnabled state I think right?

Reply:
yeah, but I think its better to seperate logic by useEffects, rather than trying to stuff everything into one useEffect and then use state to determine what code path to run. I've tested that we still are able to call stopVideoPreview when the audioVideo changes.

Reply:
Yup agree just wanted to have separate relevant ones together.